Lib/unittest/case.py

```diff
@@ -857,22 +857,28 @@ def assertAlmostEqual(self, first, second, places=None, msg=None,
             raise TypeError("specify delta or places not both")
 
         if delta is not None:
-            if abs(first - second) <= delta:
+            diff = abs(first - second)
+            if diff <= delta:
                 return
 
-            standardMsg = '%s != %s within %s delta' % (safe_repr(first),
-                                                        safe_repr(second),
-                                                        safe_repr(delta))
+            standardMsg = '%s != %s within %s delta (%s difference)' % (
+                safe_repr(first),
+                safe_repr(second),
+                safe_repr(delta),
+                safe_repr(diff))
         else:
             if places is None:
                 places = 7
 
-            if round(abs(second-first), places) == 0:
+            diff = abs(second - first)
+            if round(diff, places) == 0:
                 return
 
-            standardMsg = '%s != %s within %r places' % (safe_repr(first),
-                                                          safe_repr(second),
-                                                          places)
+            standardMsg = '%s != %s within %r places (%s difference)' % (
```

Comment:
To my understanding `delta` is entirely about acceptable *difference* (hence it makes sense to show the absolute diff) whereas `places` is about decimal precision.
I'm also not sure what's the use case for `assertNotAlmostEqual()` but whoever uses it will find this more readable than the current message:

```
self.assertNotAlmostEqual(5, 10, delta=15)

...produces:

AssertionError: 5 == 10 within 15 delta (5 difference)
```

So in summary, for `assertNotAlmostEqual`,  I'd be for showing the difference if `delta` is specified but not if `places` is specified.